File: cpp/Mod_matrikelNr.cpp
#pragma once
#include "Mod_removeDuplicates.cpp"

ListInt newList(ListInt mat, int mod){
	ListInt res;
	for(int i=0; i<length(mat); i++){
		res = append(res, sp_mod(get(mat, i), mod));
	}
	return res;
}

ListInt matrikelNr(ListInt mat){
	int mod = 10;
	while(true){
		ListInt nl = newList(mat, mod);
		if(length(removeDuplicates(nl)) != length (nl)){
			mod = mod*10;
		}else{
			return nl;
		}
	}
}
	

File: hs/Mod_matrikelNr.hs
module Mod_matrikelNr ( matrikelNr ) where
import Prelude hiding (and, or) 
import Debug.Trace
import Splib
import Types
import Mod_removeDuplicates

matrikelNr :: [Int] -> [Int]
matrikelNr list = matrikelNrHelper list 10

matrikelNrHelper :: [Int] -> Int -> [Int]
matrikelNrHelper list modNr | length (removeDuplicates newList) == length newList = newList
                            | otherwise = matrikelNrHelper list (modNr*10)
                              where
                               newList = map modX list
                               modX a = mod a modNr
